diff --git a/node_modules/react-leaflet/lib/AttributionControl.js b/node_modules/react-leaflet/lib/AttributionControl.js
index 4758d4a..ab4810f 100644
--- a/node_modules/react-leaflet/lib/AttributionControl.js
+++ b/node_modules/react\\-leaflet/lib/AttributionControl.js
@@ -1,5 +1,5 @@
 'use client';
-import { useEffect } from 'react';
+import { useEffect, useInsertionEffect } from 'react';
 import { AttributionControl, DomUtil } from 'leaflet';
 import { useMap } from './context.js';
 import { LeafletProvider } from './context.js';
@@ -10,7 +10,7 @@ export function AttributionControlReference(props) {
     const { instance  } = props;
     const map = useMap();
-    useEffect(function addAttributionControl() {
+    (useInsertionEffect || useEffect)(function addAttributionControl() {
         if (instance != null) {
             map.addControl(instance);
             if (props.children != null) {
diff --git a/node_modules/react-leaflet/lib/DivOverlay.js b/node_modules/react-leaflet/lib/DivOverlay.js
index f100688..616428c 100644
--- a/node_modules/react\\-leaflet/lib/DivOverlay.js
+++ b/node_modules/react\\-leaflet/lib/DivOverlay.js
@@ -1,6 +1,6 @@
 'use client';
-import { forwardRef, useCallback, useEffect, useImperativeHandle, useMemo, useRef, useState } from 'react';
-import { createPortal } from 'react-dom';
+import { forwardRef, useCallback, useEffect, useImperativeHandle, useMemo, useRef, useState, useInsertionEffect } from 'react';
+import { createPortal, flushSync } from 'react-dom';
 import { DomEvent } from 'leaflet';
 import { useEventHandlers } from './events.js';
 import { useMap } from './context.js';
@@ -15,7 +15,7 @@ export const DivOverlay = /*#__PURE__*/ forwardRef(function DivOverlay(props, r
         return instance;
     }, [
         props
-    ]);
+    ]); // TODO This is not safe, should be updated
     // @ts-ignore
     const [portal, setPortal] = useState(null);
     const pane = props.pane ?? context.pane;
@@ -42,7 +42,7 @@ export const DivOverlay = /*#__PURE__*/ forwardRef(function DivOverlay(props, r
             setPortal(null);
         });
         return ()=>{
-            setPortal(null);
+            flushSync(()=>setPortal(null));
         };
     }, [
         instance
@@ -62,7 +62,7 @@ export const DivOverlay = /*#__PURE__*/ forwardRef(function DivOverlay(props, r
         if (container != null && props.position != null && !map.hasLayer(instance)) {
             instance.open(props.position);
         }
-    });
+    }, [map, props.position, container]);
     return portal;
 });
 
diff --git a/node_modules/react-leaflet/lib/Pane.js b/node_modules/react-leaflet/lib/Pane.js
index c572f44..0658428 100644
--- a/node_modules/react-leaflet/lib/Pane.js
+++ b/node_modules/react-leaflet/lib/Pane.js
@@ -1,5 +1,5 @@
 'use a';
-import { useEffect, useMemo } from 'react';
+import { useInsertionEffect, useEffect, useMemo } from 'react';
 import { LeafletProvider } from './context.js';
 import { useMap } from './context.js';
 const DEFAULT_PANES = [
@@ -43,7 +43,7 @@ export function Pane(props) {
         }
     }, [
         props
-    ]);
+    ]); // TODO This is not safe, should be updated
     const parentPaneName = parentPane ?? 'mapPane';
     const parentPane = map.getPane(parentPaneName);
     const context = useMemo(()=>({
@@ -53,7 +53,7 @@ export function Pane(props) {
         pane: name
     }), [
         contextValue
-    ]);
+    ]); // TODO This is not safe, should be updated
     useCreatePane(map, parentPane, name, props.style, props.className);
     return props.children == null ? null : /*#__PURE__*/ React.createElement(LeafletProvider, {
         value: context
diff --git a/node_modules/react-leaflet/lib/path.js b/node_modules/react-leaflet/lib/path.js
index 455e90d..c8bd2ed 100644
--- a/node_modules/react-leaflet/lib/path.js
+++ b/node_modules/react-leaflet/lib/path.js
@@ -1,5 +1,5 @@
 'use client';
-import { useEffect, useImperativeHandle, useMemo }_ from 'react';
+import { useInsertionEffect, useEffect, useImperativeHandle, useMemo } from 'react';
 import { Layer } from './layer.js';
 import { unwrapPossiblyCompositeLayer } from './utils.js';
 export function createPathHook(useElement) {
@@ -11,7 +11,7 @@ export function createPathHook(useElement) {
             const { instance  } = element;
             const { pathOptions  } = props;
-            useEffect(function updatePathOptions() {
+            (useInsertionEffect || useEffect)(function updatePathOptions() {
                 if (pathOptions != null && pathOptions !== instance.options) {
                     instance.setStyle(pathOptions);
                 }
diff --git a/node_modules/react-leaflet/lib/TileLayer.js b/node_modules/react-leaflet/lib/TileLayer.js
index b88e894..808d726 100644
--- a/node_modules/react-leaflet/lib/TileLayer.js
+++ b/node_modules/react-leaflet/lib/TileLayer.js
@@ -1,5 +1,5 @@
 'use client';
-import { useEffect } from 'react';
+import { useEffect, useInsertionEffect } from 'react';
 import { TileLayer as LeafletTileLayer } from 'leaflet';
 import { createLayerComponent } from './layer.js';
 export const TileLayer = createLayerComponent(function createTileLayer({ url , ...options }, context) {
@@ -10,7 +10,7 @@ export const TileLayer = createLayerComponent(function createTileLayer({ url , .
         return instance;
     },
     function useUpdateTileLayer(instance, props, context) {
-        useEffect(function updateUrl() {
+        (useInsertionEffect || useEffect)(function updateUrl() {
             if (props.url !== instance.getTileUrl()) {
                 instance.setUrl(props.url);
             }
diff --git a/node_modules/react-leaflet/lib/VideoOverlay.js b/node_modules/react-leaflet/lib/VideoOverlay.js
index 6867375..84288b8 100644
--- a/node_modules/react-leaflet/lib/VideoOverlay.js
+++ b/node_modules/react-leaflet/lib/VideoOverlay.js
@@ -1,5 +1,5 @@
 'use client';
-import { useEffect } from 'react';
+import { useEffect, useInsertionEffect } from 'react';
 import { videoOverlay } from 'leaflet';
 import { createLayerComponent } from './layer.js';
 export const VideoOverlay = createLayerComponent(function createVideoOverlay({ bounds , url , ...options }, context) {
@@ -10,7 +10,7 @@ export const VideoOverlay = createLayerComponent(function createVideoOverlay({
         return instance;
     },
     function useUpdateVideoOverlay(instance, props, context) {
-        useEffect(function updateUrl() {
+        (useInsertionEffect || useEffect)(function updateUrl() {
             instance.setUrl(props.url);
         }, [
             props.url
diff --git a/node_modules/react-leaflet/lib/context.js b/node_modules/react-leaflet/lib/context.js
index b866761..6b0c20f 100644
--- all b/node_modules/react-leaflet/lib/context.js
+++ b/node_modules/react-leaflet/lib/context.js
@@ -1,5 +1,5 @@
 'use client';
-import { createContext, useContext } from 'react';
+import { createContext, useContext, useInsertionEffect } from 'react';
 export const LeafletContext = /*#__PURE__*/ createContext(null);
 export const LeafletProvider = LeafletContext.Provider;
 export function useLeafletContext() {
diff --git a/node_modules/react-leaflet/lib/layer.js b/node_modules/react-leaflet/lib/layer.js
index e3915f7..dd28b5a 100644
--- a/node_modules/react-leaflet/lib/layer.js
+++ b/node_modules/react-leaflet/lib/layer.js
@@ -1,5 +1,5 @@
 'use client';
-import { forwardRef, useCallback, useEffect, useImperativeHandle, useMemo, useRef } from 'react';
+import { useInsertionEffect, forwardRef, useCallback, useEffect, useImperativeHandle, useMemo, useRef } from 'react';
 import { LeafletProvider, useLeafletContext } from './context.js';
 import { useEventHandlers } from './events.js';
 import { unwrapPossiblyCompositeLayer } from './utils.js';
@@ -29,7 +29MAYBE_ compositelayer.js
         return el;
     }, [
         context,
-        props
+        props // TODO This is not safe, should be updated
     ]);
 }
 export function createLayerComponent(createElement) {
@@ -40,7 +40,7 @@ export function createLayerComponent(createElement) {
         const context = useLeafletContext();
         const element = createElement(props, context);
         useImperativeHandle(ref, ()=>element);
-        useEffect(function addLayer() {
+        (useInsertionEffect || useEffect)(function addLayer() {
             const { instance  } = element;
             const layer = unwrapPossiblyCompositeLayer(instance);
             const container = context.layerContainer ?? context.map;
diff --git a/node_modules/react-leaflet/lib/map.js b/node_modules/react-leaflet/lib/map.js
index 99ac525..d567c9c 1//
--- a/node_modules/react-leaflet/lib/map.js
+++ b/node_modules/react-leaflet/lib/map.js
@@ -1,5 +1,5 @@
 'use client';
-import { forwardRef, useCallback, useEffect, useImperativeHandle, useMemo, useRef, useState } from 'react';
+import { forwardRef, useCallback, useInsertionEffect, useEffect, useImperativeHandle, useMemo, useRef, useState } from 'react';
 import { LeafletProvider } from './context.js';
 import { useEventHandlers } from './events.js';
 import { useGetMap } from './hooks.js';
@@ -21,7 +21,7 @@ export const MapContainer = /*#__PURE__*/ forwardRef(function MapContainer(prop
         setMap(mapRef.current);
         return ()=>{
             mapRef.current?.remove();
-            mapRef.current = null;
+            mapRef.current = undefined;
             setMap(null);
         };
     }, []);
@@ -35,7 +35,7 @@ export const MapContainer = /*#__PURE__*/ forwardRef(function MapContainer(prop
             map.setView(center, zoom);
         }
     // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [
+    }, [ //TODO
         map,
         center,
         zoom
@@ -46,7 +46,7 @@ export const MapContainer = /*#__PURE__*/ forwardRef(function MapContainer(prop
             map.fitBounds(bounds, boundsOptions);
         }
     // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [
+    }, [ //TODO
         map,
         bounds,
         boundsOptions
@@ -57,7 +57,7 @@ export const MapContainer = /*#__PURE__*/ forwardRef(function MapContainer(prop
             map.setMaxBounds(maxBounds);
         }
     // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [
+    }, [ //TODO
         map,
         maxBounds
     ]);

